### PR TITLE
fix: payments on round-robin events

### LINKED ISSUE #11043: Page not Found in Round Robin Team Payment
### Issue Summary

In the Round Robin Team Event type with Stripe and Confirmation Required is enabled. The App returns a 404 when booking is submitted.

### Steps to Reproduce

1. Enable Stripe  
2. Create an event-type with the settings mentioned in summary.
3. Book an event.

[Video](https://youtu.be/kaYvNgkYFS0)

### LINKED ISSUE #11088: [CAL-2402] Payments on round robin event types
Got a report that payment on round robin events are currently not working.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2402](https://linear.app/calcom/issue/CAL-2402/payments-on-round-robin-event-types)</sub>

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Fixes round robin with payments

Fixes #11043 

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Mandatory Tasks

- [X] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/ee/payments/pages/payment.tsx [hunk 1]
```diff
@@ -106,9 +106,11 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
 
   if (!eventType) return { notFound: true };
 
-  const [user] = eventType.users;
-  if (!user) return { notFound: true };
+  if (eventType.users.length === 0 && !!!eventType.team) return { notFound: true };
```
Comment (alannnc) [inline packages/features/ee/payments/pages/payment.tsx:109]: Round robin event types has no users.

File: packages/features/ee/payments/pages/payment.tsx [hunk 2]
```diff
@@ -106,9 +106,11 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
...
 
+  const [user] = eventType?.users.length
+    ? eventType.users
+    : [{ name: null, theme: null, hideBranding: null }];
```
Comment (alannnc) [inline packages/features/ee/payments/pages/payment.tsx:113]: Defaulting to a user with null values

**Review by Udit-takkar — COMMENTED** [review]
@alannnc works fine but these { name: null, theme: null, hideBranding: null, username: null } are used in payment page. we might display wrong theme, hidebranding if we just assign null in case of RR.

**Comment by alannnc** [comment]
> @alannnc works fine but these { name: null, theme: null, hideBranding: null, username: null } are used in payment page. we might display wrong theme, hidebranding if we just assign null in case of RR.

image

As you can see above we are already sending null when the user is missing it's values. Regardless it is an scenario that shouldn't happen at all but if it happens then we don't show up anything there.
@Udit-takkar

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**